src/popup/popup.js

```diff
@@ -501,6 +502,19 @@ app.factory('NamingMask', [
       text: (mediaItem) => mediaItem.text || '',
       width: (mediaItem) => mediaItem.width || '',
       height: (mediaItem) => mediaItem.height || '',
+      // URL properties
+      hash: (mediaItem) => mediaItem.urlProps.hash || '',
+      host: (mediaItem) => mediaItem.urlProps.host || '',
+      hostname: (mediaItem) => mediaItem.urlProps.hostname || '',
+      href: (mediaItem) => mediaItem.urlProps.href || '',
+      origin: (mediaItem) => mediaItem.urlProps.origin || '',
+      password: (mediaItem) => mediaItem.urlProps.password || '',
+      pathname: (mediaItem) => mediaItem.urlProps.pathname || '',
+      port: (mediaItem) => mediaItem.urlProps.port || '',
+      protocol: (mediaItem) => mediaItem.urlProps.protocol || '',
+      search: (mediaItem) => mediaItem.urlProps.search || '',
+      searchParams: (mediaItem) => mediaItem.urlProps.searchParams || '',

```

Comment:
You work fast on these PRs. FYI, searchParams is an interface not a DOMString property.

Reply:
Just trying to help. 😄 Yeah, I just noticed it while I was updating the Wiki page. I'm not entirely sure, how this could be solved right now but I'm on it.